### DISCUSSION TITLE
Tabbing out from or into the game causes display to flicker

### DISCUSSION THREAD

**Discussion by Rian8337** [ppy/osu#17380] [Q&A] [answered]
I've had this problem ever since I tried osu!lazer, but didn't bother asking as I didn't find this annoying. However, it started to bug me out nowadays that I'm playing the game more often than osu!stable.

Unfortunately, I didn't have a good enough computer nor device to record, but I thought the [video](https://youtu.be/QdjbM8HRz7g) is enough to give an idea of what's going on (had to put it on YouTube due to GitHub file size limit).

In case it's not clear, I was simply alt-tabbing between Notepad and the game, which caused both my displays to flicker. I'm not sure if this is a problem on my end or the game, so I didn't immediately create an issue.

Logs:
[network.log](https://github.com/ppy/osu/files/8313196/network.log)
[performance-update.log](https://github.com/ppy/osu/files/8313191/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/8313192/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8313193/updater.log)
[database.log](https://github.com/ppy/osu/files/8313194/database.log)
[performance-audio.log](https://github.com/ppy/osu/files/8313188/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/8313189/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/8313190/performance-input.log)
[performance.log](https://github.com/ppy/osu/files/8313186/performance.log)

**Comment by peppy** [comment]
Can you confirm that you are playing both lazer and stable in fullscreen mode? If so, I guess you're also saying that stable handles it in a better way than lazer, correct? Are you running stable in compatibility mode or anything?

Related / duplicate: https://github.com/ppy/osu/discussions/16914

**Reply by Rian8337** [reply]
> Can you confirm that you are playing both lazer and stable in fullscreen mode?

Yes, I am playing both stable and lazer in fullscreen mode. This behavior doesn't appear when playing both games in windowed mode.

> If so, I guess you're also saying that stable handles it in a better way than lazer, correct?

I'd say yes. When replicating the action, this behavior doesn't appear in stable.

> Are you running stable in compatibility mode or anything?

No, I'm not running stable in compatibility mode.

**Reply by peppy** [reply]
Thanks. This is going to require some further investigation (as to whether it's something we're doing wrong, or an issue at SDL's end).

**Reply by Rian8337** [reply]
After further testing, I was able to replicate this behavior in stable with "render at native resolution" setting turned off, however:

1. It only occurs when I'm not using the native resolution of my monitor
2. Display flickering only occurs once per tabbing in/out instead of multiple times

I'm not sure if this is related, but I think it's worth mentioning.

**Comment by frenzibyte** [comment, marked as answer]
Escalated to #17381

**Comment by Susko3** [comment]
Try setting the resolution in osu! to the monitor native resolution (eg. change from `Default` to `1920x1080`) and see if it makes a difference.

If necessary, cycle the window modes (F11) and/or close and reopen the game.

**Reply by Rian8337** [reply]
It didn't make a difference, however I noticed another problem that I will make in a separate discussion since it's out of scope of this discussion.

**Reply by Susko3** [reply]
Could you try enabling GPU scaling in your GPU settings? (https://github.com/ppy/osu/issues/17381#issuecomment-1100207099)

**Reply by Rian8337** [reply]
I don't think the option to "enable/disable" GPU scaling is available for me. However, I tried switching between both options in the image provided and the behavior stays the same. I might be looking in the wrong settings, though.

<details>
<summary>Image</summary>

![image](https://user-images.githubusercontent.com/52914632/163611446-0f3996cd-7398-4994-aff8-30f3e78c8970.png)
</details>
